src/lib.rs

```diff
@@ -326,18 +320,14 @@ impl<T: Send> Iterator for BareSubscriber<T> {
         self.try_recv().ok()
     }
 }
-unsafe impl<T: Send> Send for BarePublisher<T> {}
-unsafe impl<T: Send> Send for BareSubscriber<T> {}
-//impl<T: Send> !Sync for BarePublisher<T>{}
-//impl<T: Send> !Sync for BareSubscriber<T>{}
 
 /// Helper struct used by sync and async implementations to wake Tasks / Threads
 #[derive(Debug)]
 pub struct Waker<T> {
     /// Vector of Tasks / Threads to be woken up.
-    pub sleepers: RefCell<Vec<Arc<T>>>,
```

Comment:
ReffCell is does not implement the Sync trait as it bypasses the borrow checker.